### ISSUE TITLE
strange log line addition causes "westnet-eastnet-auto" #1: authentication failed: no certificate matched RSASSA-PSS with SHA2_512 and '@east'

### ISSUE THREAD

**Issue by paulwouters** [libreswan/libreswan#1563]
This is really odd.

When commit e888dd15680724fa5c2fb3d257ebf4ef67818338 is in the tree, a few test cases such as ikev2-delete-05-sa-start  suddenly fail to find their RSA key ?

```
"westnet-eastnet-auto" #5: processing decrypted IKE_AUTH request: SK{IDi,IDr,AUTH,SA,TSi,TSr}
"westnet-eastnet-auto" #5: authentication failed: no certificate matched RSASSA-PSS with SHA2_512 and '@east'
```

I have no idea how these two things can be related.

**Comment by cagney** [comment]
Until this can be explained away e888dd15680724fa5c2fb3d257ebf4ef67818338 should be reverted.

**Comment by letoams** [comment]
On Jan 21, 2024, at 09:58, cagney ***@***.***> wrote:﻿
Until this can be explained away e888dd1 should be reverted.I didn’t because I wanted to be sure the situation doesn’t look get overlooked. Not sure reverting serves much of a purpose ?Message ID: ***@***.***>

**Comment by cagney** [comment]
> I didn’t because I wanted to be sure the situation doesn’t look get overlooked.

With two bugs tracking it, how could it be overlooked?

> Not sure reverting serves much of a purpose ?

I'll assume this question is rhetorical.

**Comment by cagney** [comment]
I've pulled e888dd15680724fa5c2fb3d257ebf4ef67818338 et.al. so I have a clean baseline for testing #1554
I've also patched up  be1b45921a0a5dfae2c7f26f108404374935eb96 and pushed it with a test update.  The result should be a test run with zero failures (at least it was here).

Adding a connection with keys requires two transactions: first add the connection, and second add the keys.  This trips up when, while adding the connection, a message with `RC_LOG_SERIOUS` is logged as whack will treat that as fatal and immediately detach (leaving no keys).  Once I've a clean test run using RC_LOG, the reverted changes can go back in.
